File: 251004/Yelkin/4-5/CommentService/src/main/java/by/yelkin/CommentService/mapper/CommentMapper.java
package by.yelkin.CommentService.mapper;

import by.yelkin.CommentService.dto.CommentRequestTo;
import by.yelkin.CommentService.dto.CommentResponseTo;
import by.yelkin.CommentService.entity.Comment;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface CommentMapper {
    Comment toEntity(CommentRequestTo request);
    CommentResponseTo toResponse(Comment comment);
    List<CommentResponseTo> toPostResponseList(Iterable<Comment> comments);

}

File: 251003/Kukhatskavolets/microblog/src/main/kotlin/by/kukhatskavolets/mappers/CommentMapper.kt
package by.kukhatskavolets.mappers

import by.kukhatskavolets.dto.requests.CommentRequestTo
import by.kukhatskavolets.dto.responses.CommentResponseTo
import by.kukhatskavolets.entities.Comment
import by.kukhatskavolets.entities.Tweet


fun CommentRequestTo.toEntity(tweet: Tweet): Comment {
    return Comment(
        id = this.id,
        tweet = tweet,
        content = this.content,
    )
}

fun Comment.toResponse(): CommentResponseTo {
    return CommentResponseTo(
        id = this.id,
        tweetId = this.tweet.id,
        content = this.content,
    )
}
